runtime-template-client/runtime_template_client.go

```diff
@@ -0,0 +1,200 @@
+package runtimetemplateclient
```

Comment:
This is something @pritt20 and I discussed. As far as I know, we don't currently list `git` as a prerequisite. In this particular instance, we're trying to get the user up and running with the least amount of dependencies and prerequisites as possible. Even though most of our user base will probably have `git` installed, it just felt too risky to rely on for such a critical onboarding step where we're doing everything we can to optimize user experience and remove friction. 

Are there other places in the CLI code where `git` is currently a hard dependency?

Reply:
Yeah that is true, we do currently use `git` in a few places but not as a hard dependency. If we go with the tar/gzip approach can we follow a similar structure to what I listed above, so we do not need a new top-level client? Where a top-level variable can be replaced by a unit test to simulate the file download. We do this in a few other places, e.g. for [deploying an image](https://github.com/astronomer/astro-cli/blob/40db395aa829e50a4dc0781d626365fd80c23da6/cmd/cloud/deploy.go#L41) which a unit test cannot really do.

I'm also hoping we can extract the contents straight into the target dir instead of into a temp dir and then copying the dirs and files across?

Reply:
Thanks for the suggestion @jeremybeard and @schnie.

I have updated the code to reflect above changes.

I tried to dump the extracted files to target directory but the tarball that is extracted contains all templates directories and use path like `target-directory/astronomer-templates-7a8aa17/<templates-directories>`. We want the selected template directory files to be in `target-directory` without any sub directories. Therefore I have kept the temp directory logic intact to move files around and copy only required files to target directory 😅